src/App.tsx

```diff
@@ -1,93 +1,112 @@
-              <label htmlFor="toggle-view2">1234567890</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-        </ul>
-      </section>
+  const handleAddTodo = (newTodo: Todo) => setTodos([...todosList, newTodo]);
+
+  const handleDeleteTodo = (todoId: number) => {
+    setTodos(todosList.filter(todo => todo.id !== todoId));
+    setCompletedTodosId(completedTodosId.filter(number => number !== todoId));
+  };
 
-      <footer className="footer">
-        <span className="todo-count" data-cy="todosCounter">
-          3 items left
-        </span>
+  const deleteCompletedTodos = () => {
+    setCompletedTodosId([]);
+    setTodos(todosList.filter(todo => !todo.completed));
+  };
 
-        <ul className="filters">
-          <li>
-            <a href="#/" className="selected">All</a>
-          </li>
+  const handleSetSortBy = (value: SortBy) => {
+    setSortBy(value);
+  };
 
-          <li>
-            <a href="#/active">Active</a>
-          </li>
+  const handleSetTodos = (newTodos: Todo[]) => {
+    setTodos(newTodos);
+  };
 
-          <li>
-            <a href="#/completed">Completed</a>
-          </li>
-        </ul>
+  const updateCheckAllTodo = () => {
+    if (todosList.length - completedTodosId.length === 0) {
+      setTodos(todosList.map(todo => ({ ...todo, completed: false })));
+      setCompletedTodosId([]);
+    } else {
+      setTodos(todosList.map(todo => ({ ...todo, completed: true })));
+      setCompletedTodosId(todosList.map(todo => todo.id));
+    }
+  };
 
-        <button type="button" className="clear-completed">
-          Clear completed
-        </button>
-      </footer>
+  return (
+    <div className="todoapp">
+      <Header handleAddTodo={handleAddTodo} />
+
+      <section className="main">
+        <input
+          type="checkbox"
+          checked={todosList.length - completedTodosId.length === 0}
+          id="toggle-all"
+          className="toggle-all"
+          data-cy="toggleAll"
+          onClick={updateCheckAllTodo}
+        />
+        {todosList.length > 0 && (
+          <label htmlFor="toggle-all">Mark all as complete</label>
+        )}
+        {todosList.length !== 0 && (
+          <ul className="todo-list" data-cy="todosList">
+            <TodoList
+              todos={todosList}
+              handleUpdateCheckTodo={handleUpdateCheckTodo}
+              handleDeleteTodo={handleDeleteTodo}
+              sortBy={sortBy}
+              handleSetTodos={handleSetTodos}
+            />
+          </ul>
+        )}
+      </section>
+      {todosList.length !== 0 && (
+        <TodosFilter
+          handleSetSortBy={handleSetSortBy}
+          sortBy={sortBy}
+          completedTodosId={completedTodosId}
+          deleteCompletedTodos={deleteCompletedTodos}
+          todosList={todosList}
+        />
+      )}
```

Comment:
you can combine it in one condition

Reply:
I can not do it, becouse TodoList is in "section" element and TodosFilter is not in "section".

Reply:
oh, yeah, sorry